internal/hcp/internal/types/link.go

```diff
@@ -63,7 +69,28 @@ func validateLink(res *DecodedLink) error {
 			Name:    "resource_id",
 			Wrapped: resource.ErrMissing,
 		})
+	} else {
+		_, parseErr := hcpresource.FromString(res.Data.ResourceId)
+		if parseErr != nil {
+			err = multierror.Append(err, resource.ErrInvalidField{
+				Name:    "resource_id",
+				Wrapped: errInvalidHCPResourceID,
+			})
+		}
 	}
 
 	return err
 }
+
+func GenerateTestResourceID(t *testing.T) string {
```

Comment:
Maybe we should put this in `link_test.go` since it's only for test code?

Reply:
Test methods aren't exported, so we wouldn't be able to reuse this method if it was in `link_test.go`. I can move it to its own `testing.go` file, though, which is what we've done in other packages!